homeassistant/components/ultrasync/config_flow.py

```diff
@@ -0,0 +1,123 @@
+"""Config flow for the Interlogix/Hills ComNav UltraSync Hub."""
+import logging
+from typing import Any, Dict, Optional
+
+import ultrasync
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PIN,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .const import DEFAULT_NAME, DEFAULT_SCAN_INTERVAL
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AuthFailureException(Exception):
+    """A general exception we can use to track Authentication failures."""
+
+    pass
+
+
+def validate_input(hass: HomeAssistantType, data: dict) -> Dict[str, Any]:
+    """Validate the user input allows us to connect."""
+
+    usync = ultrasync.UltraSync(
+        host=data[CONF_HOST], user=data[CONF_USERNAME], pin=data[CONF_PIN]
+    )
+
+    # validate by attempting to authenticate with our hub
+
+    if not usync.login():
+        # report our connection issue
+        raise AuthFailureException()
+
+    return True
```

Comment:
We don't need to return anything. On failure this function will raise an exception.